### ISSUE TITLE
[Bug]: multiple open msgpackr processes after terminating a node.js process that uses bullmq

### ISSUE THREAD

**Issue by HosseinAgha** [taskforcesh/bullmq#2178]
### Version

v4.8.0

### Platform

NodeJS

### What happened?

After closing a node process that uses BullMQ multiple `msgpackr` processes remain open and consume a lot of cpu. The only dependency on `msgpackr` in my project is from BullMQ.  

![Screenshot 2023-09-12 at 5 00 52 PM](https://github.com/taskforcesh/bullmq/assets/12998499/a629b1a7-716c-488f-8498-f4c0d79cd9ff)

All the node processes details are the same as above:  
![Screenshot 2023-09-12 at 5 04 46 PM](https://github.com/taskforcesh/bullmq/assets/12998499/a129a7f1-d10c-414a-9b20-48a7f521738f)

### How to reproduce.

Use the latest version of bullmq and make it so that the Redis connection fails and ioredis tries to retry the connection.
After closing the node.js process using ctrl+c, many unused `msgpackr` processes will not terminate and start to consume all of your system's resources.  

I provide a single ioredis connection to all of my queues and workers.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by manast** [comment]
Are you sure about this? reason for asking is that in Unix, if the parent process is killed, its children are killed automatically, so it is strange that you still have node processes running after killing the parent.

**Comment by HosseinAgha** [comment]
Yes, my basic understanding was the same as yours but this is the situation I'm dealing with.  
I use a nodemon-like process manager to restart the server automatically but my server will not restart when connection to Redis fails so I don't think nodemon is the issue. I just found out that there will be orphan processes with `msgpackr` running regardless of the Redis connection status.  
My main suspect is `msgpackr` as it is a native node.js C++ addon (binary) and these addons tend to generate such issues. Somehow SIGKILL does not reach the processes created by `msgpackr`.  
I even suspect that there may be some malware in `msgpackr` trying to mine crypto or something else. 😄    
I'll investigate further when I have more free time.

**Comment by manast** [comment]
What makes you believe msgpackr is creating this issue? BTW; it is just plain javascript: https://www.npmjs.com/package/msgpackr

**Comment by HosseinAgha** [comment]
I don't think so. Please take a look at the screenshot. `msgpackr` depends on `msgpackr-extract` package which is in binary format specifically: `msgpackr-extract-darwin-arm64/node.napi.glibc.node`.

**Comment by manast** [comment]
I see, it states that it is an optional dependency but it is enabled by default: https://www.npmjs.com/package/msgpackr

**Comment by manast** [comment]
Are you using sandboxed processors or standard?

**Comment by HosseinAgha** [comment]
We are using the standard processors.
BTW sorry for the delayed response.

**Comment by manast** [comment]
In order to be able to investigate this issue further we would need some simple test that we can run to reproduce it.

**Comment by HosseinAgha** [comment]
Actually, I don't know a good way to provide you with a MacOS sandbox. I can confirm this only happens in MacOS.

**Comment by manast** [comment]
@HosseinAgha I am on Mac. What piece of test code should I use to reproduce it?
